### PR TITLE
fix(datatable/quickadd): open detail view on AddandOpen Button

### PR DESCRIPTION
**Pull request by sahil143**
This PR introduces AddandOpen Button functionality.
Adds the workItem and then redirect to detail page of newly added workItem

### CODE REVIEW

File: src/app/components/work-item-quick-add/work-item-quick-add.component.ts [hunk 1]
```diff
@@ -272,11 +272,11 @@ export class WorkItemQuickAddComponent implements OnInit, OnDestroy, AfterViewIn
             this.createLinkObject(this.parentWorkItemId, workItem.id, '25c326a7-6d03-4f5a-b23b-86a9ee4171e9');
             let tempLinkObject = {'data': this.linkObject};
             this.workItemService.createLink(tempLinkObject)
-              .subscribe(([link, includes]) => {
-                this.workItemService.emitAddWIChild(this.parentWorkItemId);
+              .subscribe(([link, includes]) => {
```
Comment (sanbornsen) [inline src/app/components/work-item-quick-add/work-item-quick-add.component.ts]: There are extra empty spaces at the end of the line.

File: src/app/components/work-item-quick-add/work-item-quick-add.component.ts [hunk 2]
```diff
@@ -272,11 +272,11 @@ export class WorkItemQuickAddComponent implements OnInit, OnDestroy, AfterViewIn
             this.createLinkObject(this.parentWorkItemId, workItem.id, '25c326a7-6d03-4f5a-b23b-86a9ee4171e9');
             let tempLinkObject = {'data': this.linkObject};
             this.workItemService.createLink(tempLinkObject)
-              .subscribe(([link, includes]) => {
-                this.workItemService.emitAddWIChild(this.parentWorkItemId);
+              .subscribe(([link, includes]) => {                
+                this.workItemService.emitAddWIChild({pwid: this.parentWorkItemId, wid: workItem.id, status: openStatus});
```
Comment (sanbornsen) [inline src/app/components/work-item-quick-add/work-item-quick-add.component.ts]: Break this line

File: src/app/services/work-item.service.ts [hunk 3]
```diff
@@ -63,8 +63,8 @@ export class WorkItemService {
 
   private selfId;
 
-  public addWIObservable: Subject<WorkItem> = new Subject();
-  public addWIChildObservable: Subject<string> = new Subject();
+  public addWIObservable: Subject<any> = new Subject();
```
Comment (sanbornsen) [inline src/app/services/work-item.service.ts]: Why do you have to type it to ```any```?

File: src/app/services/work-item.service.ts [hunk 4]
```diff
@@ -63,8 +63,8 @@ export class WorkItemService {
...
+  public addWIChildObservable: Subject<any> = new Subject();
```
Comment (sanbornsen) [inline src/app/services/work-item.service.ts]: Why do you have to type it to any?

**Comment by Raunak1203** [comment]
@sahil143 add and open works but order of execution changes after Add and open:
Steps:
-Create a new work item
-And click on Add and open
-Detail page is opened
-Now click cancel.
-Newly created workitem is not displayed on top

**Comment by sahil143** [comment]
@Raunak1203 
>add and open works but order of execution changes after Add and open:
>Steps:
>-Create a new work item
>-And click on Add and open
>-Detail page is opened
>-Now click cancel.
>-Newly created workitem is not displayed on top

This change is not introduced by this PR. Even when a workitem is created by clicking on `add` button and then refresh the page the order of execution changes(workitem is not displayed on top).

**Comment by fabric8cd** [comment]
@sahil143 fabric8/fabric8-ui:SNAPSHOT-PR-2404-4 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2404-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@sahil143 fabric8/fabric8-ui:SNAPSHOT-PR-2404-5 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2404-fabric8-planner.badger.fabric8.io

**Pull request merged**